src/main/java/teammates/ui/webapi/action/GetHasResponsesAction.java

```diff
@@ -0,0 +1,57 @@
+package teammates.ui.webapi.action;
+
+import org.apache.http.HttpStatus;
+
+import teammates.common.datatransfer.attributes.FeedbackQuestionAttributes;
+import teammates.common.datatransfer.attributes.FeedbackSessionAttributes;
+import teammates.common.util.Const;
+import teammates.ui.webapi.output.HasResponsesData;
+
+/**
+ * Checks whether a session has responses.
+ */
+public class GetHasResponsesAction extends Action {
+
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return AuthType.LOGGED_IN;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        // Anyone can check whether a course has responses
+
+        String questionId = getRequestParamValue(Const.ParamsNames.FEEDBACK_QUESTION_ID);
+        if (questionId != null) {
+            FeedbackQuestionAttributes feedbackQuestionAttributes = logic.getFeedbackQuestion(questionId);
+            FeedbackSessionAttributes feedbackSession = logic.getFeedbackSession(
+                    feedbackQuestionAttributes.getFeedbackSessionName(),
+                    feedbackQuestionAttributes.getCourseId());
+
+            gateKeeper.verifyAccessible(
+                    logic.getInstructorForGoogleId(feedbackQuestionAttributes.getCourseId(), userInfo.getId()),
```

Comment:
What happens if request with both `questionId` and `courseId` ?  It might be results in an inconsistent case where in accessControl, we checked question-id, but in execution, we returned for course. Can we make it consistent? (ignore courseId if questionId is present), and maybe add a test case for it?

Reply:
Good point! I've updated it to support both courseID and questionID; prefer questionID if both is provided.